### DISCUSSION TITLE
is it possible to make a delete api with query ?

### DISCUSSION THREAD

**Discussion by dunods** [strapi/strapi#6162] [Legacy Topics]
Hi everyone

how do i perform delete/ is it possible to delete  /localhost:1337/products?type=old

did try to customize both controller and services but it seems not working.

and is there any method to delete child within the relationship from the api ?
any idea?

**Comment by petnd** [comment]
Hi, I would recommend something like this:

Create a new route
(example for "config" model)
```javascript
{
      "method": "DELETE",
      "path": "/configs/",
      "handler": "config.deleteQuery",
      "config": {
        "policies": []
      }
    }
```


Then you need to write your own controller "deleteQuery". It should look something like this:




```javascript
async deleteQuery(ctx) {
    let entities;
    if (ctx.query._q) {
      entities = await strapi.services.config.search(ctx.query);
    } else {
      entities = await strapi.services.config.find(ctx.query);
    }
const deletedE = await strapi.services.config.delete({ entities[0].id });
    return sanitizeEntity(deletedE , { model: strapi.models.config});
  }
````

it´s not good code, but it shoudl give you an idea how to do it. Would be good to check if some object was found before deleting it.

**Reply by dunods** [reply]
somehow i did try with those method but it returned error cause sanitizeEntity is not a function .

any clue?

**Reply by derrickmehaffy** [reply]
You are going to need to loop through the response from the find unless you only want to delete the first entry

**Comment by petnd** [comment]
You need to import it first. See the docs on customizing controllers
